## Student Performance Indicator

 Life cycle of Machine Learning Project

 .understanding the problem statement  
 .Data collection
 .Data Checks to Perform
 .Exploratory Data Analysis
 .Data Pre-Processing
 .Model Training 
 .Choose best Model 


## 1) Problem Statement
.This project understands how the students performance(test score)is affected by other variables such as Gender, Ethenicity , Parental level of education, Lunch and Test Preparation course.

## 2) Data Collection
. Data Source - https://www.kaggle.com/datasets/spscientist/students-performance-in-exams?datasetId=74977
.The data consist of 8 column and 1000 rows.

## 2.1 Import Data and Required packages

Importing pandas ,Numpy,Matplotlib,Seaborn, and warnings library. 

In [38]:
import numpy as np 
import pandas as pd 
import seaborn as sns   
import matplotlib.pyplot as plt 
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

Import the csv Data as pandas dataframe

In [39]:
df = pd.read_csv('StudentsPerformance.csv')

Show Top 5 Records

In [40]:
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

Shape of the dataset

In [41]:
df.shape

(1000, 8)

# 2.2 Dataset Information

gender : sex of students -> (Male/female)
race/ethnicity : ethnicity of students -> (Group A, B,C, D,E)
parental level of education : parents' final education ->(bachelor's degree,some college,master's degree,associate's degree,high school)
lunch : having lunch before test (standard or free/reduced)
test preparation course : complete or not complete before test
math score
reading score
writing score



# 3 Data Checks to perform
Check Missing values
Check Duplicates
Check data type
Check the number of unique values of each column
Check statistics of data set
Check various categories present in the different categorical column

 3.1  Checking Missing Values 

In [42]:
df.isna().sum()

gender                         0
race/ethnicity                 0
parental level of education    0
lunch                          0
test preparation course        0
math score                     0
reading score                  0
writing score                  0
dtype: int64

3.2 check duplicate values

In [43]:
df.duplicated().sum()

0

There are no Duplicate Values in The dataset 

3.3 check Data Type 

In [44]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


3.4 checking the unique values in each column

In [45]:
df.nunique()

gender                          2
race/ethnicity                  5
parental level of education     6
lunch                           2
test preparation course         2
math score                     81
reading score                  72
writing score                  77
dtype: int64

3.5 checking statastic values of dataset

In [46]:
df.describe().T 

count    mean        std   min    25%   50%   75%    max
math score     1000.0  66.089  15.163080   0.0  57.00  66.0  77.0  100.0
reading score  1000.0  69.169  14.600192  17.0  59.00  70.0  79.0  100.0
writing score  1000.0  68.054  15.195657  10.0  57.75  69.0  79.0  100.0

# 3.6 Insight 
1 -From above description of numerical data, all means are very close to each other - between 66 and 68.05;
2 -All standard deviations are also close - between 14.6 and 15.19;
3 -While there is a minimum score 0 for math, for writing minimum is much higher = 10 and for reading myet higher = 17

# 3.7 Exploring Data

In [47]:
df.head(5)

gender race/ethnicity parental level of education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test preparation course  math score  reading score  writing score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

In [48]:
print("Categories in 'gender' variable:     ",end=" " )
print(df['gender'].unique())


Categories in 'gender' variable:      ['female' 'male']
